### PR TITLE
Bug: Fix retraction issues with PA pattern calibration

### PR DESCRIPTION
**Pull request by igiannakas**
Possible fix for #3154 and #3031 

Also fixed issue due to retract on layer change being enabled causing two de-retractions when starting the handle cube. And also two retractions when starting the PA lines which results in the first line (or more) being not extruded.

### CODE REVIEW

**Comment by igiannakas** [comment]
Fixed the second bug with the dual retractions and deretractions.

@SoftFever PR ready for review. This needs a second pair of eyes to cross check it, with both relative and absolute extrusion enabled.

File: src/slic3r/GUI/Plater.cpp [hunk 1]
```diff
@@ -8843,9 +8843,14 @@ void Plater::_calib_pa_pattern(const Calib_Params& params)
     changed_objects({ 0 });
     _calib_pa_select_added_objects();
 
-    const DynamicPrintConfig& printer_config = wxGetApp().preset_bundle->printers.get_edited_preset().config;
+    DynamicPrintConfig& printer_config = wxGetApp().preset_bundle->printers.get_edited_preset().config;
     DynamicPrintConfig& print_config = wxGetApp().preset_bundle->prints.get_edited_preset().config;
+    auto filament_config = &wxGetApp().preset_bundle->filaments.get_edited_preset().config;
     double nozzle_diameter = printer_config.option<ConfigOptionFloats>("nozzle_diameter")->get_at(0);
+    filament_config->set_key_value("filament_retract_when_changing_layer", new ConfigOptionBools{false});
```
Comment (SoftFever) [inline src/slic3r/GUI/Plater.cpp:8850]: I'm experiencing crashes caused by this line.
Slice the PA pattern then try to create a new project, it crashes.

File: src/libslic3r/calib.cpp [hunk 2]
```diff
@@ -572,6 +572,7 @@ void CalibPressureAdvancePattern::generate_custom_gcodes(const DynamicPrintConfi
             gcode << "; start pressure advance pattern for layer\n";
 
             gcode << m_writer.travel_to_z(layer_height, "Move to layer height");
+            gcode << m_writer.reset_e();
```
[the remark this answers is not in the document]
Comment (SoftFever) [reply src/libslic3r/calib.cpp:575]: Pls ignore this comment, I realized you were solving different retraction issue in this PR.

[the remark this answers is not in the document]
Comment (igiannakas) [reply src/libslic3r/calib.cpp:575]: Ah no worries!! Still need to figure out the crash issue....

File: src/slic3r/GUI/Plater.cpp (hunk 1 above)
Comment (igiannakas) [reply src/slic3r/GUI/Plater.cpp]: fixed

**Comment by igiannakas** [comment]
crash fixed. ready for review :)

**Review by SoftFever — APPROVED** [review]
Look good

**Pull request merged**